clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -527,6 +527,29 @@ public void testConnectDisconnectDuringInSinglePoll() throws Exception {
         control.verify();
     }
 
+    /**
+     * Tests that channel references are not leaked when channels are closed regardless
+     * of whether channel has been muted. Channels hold a reference to the last send.
+     * Test checks that channels created with large sends can be created and closed
+     * without running out of memory.
+     */
+    @Test
+    public void testChannelReferencesCleared() throws Exception {
```

Comment:
Can we measure how long this and the other test take? It seems like we are allocating 10 GB in this test alone.

Reply:
@ijuma Thanks for the review. From my last test run on my laptop, this one took 4.7 seconds and the other one took 23.6 seconds! I thought the tests were useful to make sure we can recreate the OOMs. But I don't think either of them really adds much value relative to the time that they take to run.  Perhaps it is sufficient to check that we have cleared all the maps and sets in `Selector` in all the existing `SelectorTest` tests. And I could remove the new `FetchRequestTest`. What do you think?

Reply:
That sounds good for the SelectorTest. The FetchRequestTest seems useful. Could we reduce the allocations and still trigger the OOM? Another option would be to check the memory used before and after (with GC calls in between) with a large tolerance. Not sure if that's faster.

Reply:
Modified to check memory used, my last test run took ~8 seconds and around half of that is  set up and teardown.

Reply:
One of the PR builds failed even with a tolerance of 200M (the diff was nearly 400M). So I am going to revert to checking for OOM. It adds an extra 3 seconds, but should be more reliable.

Reply:
Updated `FetchRequestTest` to trigger OOM rather than check free mem. The updated test took 9.3 seconds on my laptop.

Reply:
That sounds good.